### ISSUE TITLE
4.1.0 white list

### ISSUE THREAD

**Issue by dzz10** [tesseract-ocr/tesseract#2760]
![1](https://user-images.githubusercontent.com/47780787/68741718-57854900-0629-11ea-9459-35727917c98e.png)

m_api.SetVariable("tessedit_char_whitelist", u8"孤知道了"); or m_api.SetVariable("tessedit_char_whitelist", "孤知道了");

It does not work properly and the result is an empty string.
When I commented out the code, it worked fine.

**Comment by dzz10** [comment]
The initialization code looks like this:
m_api.Init("E:\\test\x64Debug\tessdata", "chi_sim", tesseract::OEM_LSTM_ONLY);

**Comment by dzz10** [comment]
![捕获1](https://user-images.githubusercontent.com/47780787/68742166-6e786b00-062a-11ea-8aa0-768a5a250357.PNG)

**Comment by bertsky** [comment]
Confirmed. With `chi_sim.traineddata` from https://github.com/tesseract-ocr/tessdata_best:

```sh
tesseract 2760.png stdout -l chi_sim -c tessedit_char_whitelist=孤知道了 --oem 1
Warning: Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 217

tesseract build-release/2760.png stdout -l chi_sim --oem 1 
Warning: Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 217
弧 知道 了
```

**However**, with the model from https://github.com/tesseract-ocr/tessdata_fast/, I get:

```sh
tesseract 2760.png stdout -l chi_sim -c tessedit_char_whitelist=孤知道了 --oem 1
Warning: Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 217
知道了
tesseract build-release/2760.png stdout -l chi_sim --oem 1 
Warning: Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 217
弧知道了
```

Clearly, this is related to https://github.com/tesseract-ocr/tessdata_fast/80d92b7db61cb6ddc519049990be34e4c913566f:
```
Fix extra intra-word spaces by adding config file
    
    with preserve_interword_spaces 1
```

@Shreeshrii So I guess we need that same fix for the tessdata_best models, right?

@dzz10, as for the suboptimal result with tessdata_fast, where the first character is missing completely when whitelisting is enabled: This is to be expected. Whitelisting will give empty results when the requested characters cannot be detected with high enough confidence. (And the TSV output shows a very low confidence of 37% for the "best" (but still wrong) choice `弧` when running unconstrained.)

In principle, you could try to get the decoder to try deeper beam alternatives (using `lstm_choice_mode=2` and `lstm_choice_iterations>0`). But this does not work (for some reason) in your case.

**Comment by dzz10** [comment]
tks,It's the old version

**Comment by bertsky** [comment]
@dzz10 why are you closing this? This is _not_ fixed in the current `tessdata_best` models.

**Comment by dzz10** [comment]
@bertsky click wrong

**Comment by dzz10** [comment]
Hello, has this problem been solved

**Comment by amitdo** [comment]
https://github.com/tesseract-ocr/tessdata_best/commit/5075f27776cd

@bertsky, can we close the issue?

**Comment by bertsky** [comment]
> [tesseract-ocr/tessdata_best@5075f27](https://github.com/tesseract-ocr/tessdata_best/commit/5075f27776cd)
> 
> can we close the issue?

@amitdo, no, afraid not. I still get the behaviour described above on the current master of `tessdata_best` and `tessdata_fast`.
